ffmpeg/ffmpeg.go

```diff
@@ -512,32 +529,42 @@ func InitFFmpeg() {
 	InitFFmpegWithLogLevel(FFLogWarning)
 }
 
-func ReleaseFFmpegDetectorProfile() {
-	C.lpms_dnnrelease()
-}
-func InitFFmpegWithDetectorProfile(detector DetectorProfile, deviceids string) error {
-
+func NewTranscoderWithDetector(detector DetectorProfile, deviceid string) (*Transcoder, error) {
 	switch detector.Type() {
 	case SceneClassification:
 		detectorProfile := detector.(*SceneClassificationProfile)
+		sess_config := deviceidToConfigProto(deviceid)
 		dnnOpt := &C.lvpdnn_opts{
-			modelpath:  C.CString(detectorProfile.ModelPath),
-			inputname:  C.CString(detectorProfile.Input),
-			outputname: C.CString(detectorProfile.Output),
-			deviceids:  C.CString(deviceids),
+			modelpath:   C.CString(detectorProfile.ModelPath),
+			inputname:   C.CString(detectorProfile.Input),
+			outputname:  C.CString(detectorProfile.Output),
+			sess_config: C.CString(sess_config),
 		}
 		defer C.free(unsafe.Pointer(dnnOpt.modelpath))
 		defer C.free(unsafe.Pointer(dnnOpt.inputname))
 		defer C.free(unsafe.Pointer(dnnOpt.outputname))
-		defer C.free(unsafe.Pointer(dnnOpt.deviceids))
-
-		ret := int(C.lpms_dnninit(dnnOpt))
-		if 0 != ret {
-			glog.Error("lpms_dnninit Return : ", ErrorMap[ret])
-			return ErrDNNInitialize
+		defer C.free(unsafe.Pointer(dnnOpt.sess_config))
+		handle := C.lpms_transcode_new_with_dnn(dnnOpt)
+		if handle != nil {
+			return &Transcoder{
+				handle: handle,
+				mu:     &sync.Mutex{},
+			}, nil
 		}
 	}
+	return nil, ErrDNNInitialize
+}
 
-	InitFFmpegWithLogLevel(FFLogWarning)
-	return nil
+func deviceidToConfigProto(deviceid string) string {
+	config_proto := &pb.ConfigProto{GpuOptions: &pb.GPUOptions{AllowGrowth: true, VisibleDeviceList: deviceid}}
+	bytes, err := proto.Marshal(config_proto)
+	if err != nil {
+		glog.Errorf("Unable to convert deviceid %v to Tensorflow config protobuf\n", err)
+		return ""
+	}
+	sess_config_option := "sess_config=0x"
+	for i := len(bytes) - 1; i >= 0; i-- {
```

Comment:
Is this supposed to parse each byte in `bytes` in reverse order? If so, why do we need to do that?

Reply:
Yeah this was a pain - I'll add a comment to make this clear.

But somehow FFmpeg tensorflow backend expects the bytes in the reverse order - and their suggested python script even explicitly reverse the bytes in the serialized list: 

https://github.com/livepeer/FFmpeg/blob/c83669bfc0fb683b651b3c354e3f59b736091ef2/libavfilter/dnn/dnn_backend_tf.c#L220

If this wasn't the case we could have used `hex.EncodeToString(bytes)` but oh well :/

Reply:
Looks like they re-reverse it to construct into a string that `TF_SetConfig` can actually accept. https://github.com/livepeer/FFmpeg/blob/c83669bfc0fb683b651b3c354e3f59b736091ef2/libavfilter/dnn/dnn_backend_tf.c#L251

I'll probably send a patch upstream to make this simpler.